modules/platform/nuxeo-platform-filemanager/src/main/java/org/nuxeo/ecm/platform/filemanager/service/FileManagerService.java

```diff
@@ -121,82 +120,78 @@
-    }
+    protected void initConfiguration() {
+        fileImporters = null;
+        folderImporters = null;
+        creationContainerListProviders = null;
 
-    @Override
-    public void unregisterContribution(Object contribution, String xp, ComponentInstance component) {
-        if (PLUGINS_EP.equals(xp)) {
-            xp = computePluginsExtensionPoint(contribution.getClass());
-        }
-        super.unregisterContribution(contribution, xp, component);
+        unicityEnabled = false;
+        digestAlgorithm = "sha-256";
+        fieldsXPath = null;
+        computeDigest = false;
+
+        defaultVersioningOption = DEF_VERSIONING_OPTION;
+        versioningAfterAdd = DEF_VERSIONING_AFTER_ADD;
     }
 
     @Override
     public void start(ComponentContext context) {
-        super.start(context);
-
-        registerFileImporters();
-        registerFolderImporters();
-        registerCreationContainerListProviders();
+        initConfiguration();
+        registerPlugins();
         registerUnicity();
         registerVersioning();
     }
 
-    protected void registerFileImporters() {
-        String xp = computePluginsExtensionPoint(FileImporterDescriptor.class);
-        fileImporters = getDescriptors(xp).stream()
-                                          .map(FileImporterDescriptor.class::cast)
-                                          .map(FileImporterDescriptor::newInstance)
-                                          .collect(Collectors.toMap(FileImporter::getName, Function.identity()));
-    }
-
-    protected void registerFolderImporters() {
-        String xp = computePluginsExtensionPoint(FolderImporterDescriptor.class);
-        folderImporters = getDescriptors(xp).stream()
-                                            .map(FolderImporterDescriptor.class::cast)
-                                            .map(FolderImporterDescriptor::newInstance)
-                                            .collect(Collectors.toList());
+    @Override
+    public void stop(ComponentContext context) throws InterruptedException {
+        initConfiguration();
     }
 
-    protected void registerCreationContainerListProviders() {
-        String xp = computePluginsExtensionPoint(CreationContainerListProviderDescriptor.class);
-        creationContainerListProviders = getDescriptors(xp).stream()
-                                                           .map(CreationContainerListProviderDescriptor.class::cast)
-                                                           .map(CreationContainerListProviderDescriptor::newInstance)
-                                                           .collect(Collectors.toList());
+    protected void registerPlugins() {
+        FileManagerRegistry registry = getExtensionPointRegistry(PLUGINS_EP);
+        fileImporters = registry.getContributionValues(FileImporterDescriptor.class)
+                                .stream()
+                                .map(FileImporterDescriptor.class::cast)
```

Comment:
Is the cast needed? `getContributionValues` should returned objects with the proper type already.

Reply:
It is needed, i cannot find any way of using `this.<FileImporter>` in this case.
A working alternative would be:
```
List<FileImporterDescriptor> fdescs = registry.getContributionValues(FileImporterDescriptor.class);
fileImporters = fdescs.stream()
                      .map(FileImporterDescriptor::newInstance)
                      .collect(Collectors.toMap(FileImporter::getName, Function.identity()));
```
Do you prefer this way or have a better alternative?

Reply:
Keep what you have if it works, it's not very important.

Reply:
I'd actually be curious to understand why it does not work in this case, as the custom registry API is the same than the usual helper method (and my eclipse compile does not accept use of `this.<FileImporter>` here, I don't see the logic)

Reply:
`this.<FileImporterDescriptor>` you mean?

Reply:
Yes

Reply:
Actually compile complains on "." of `registry.`, i guess that's why (usually helper method is called directly)

Reply:
I'll test once you push and I can have it in my IDE